bindgen/src/gen_go/mod.rs

```diff
@@ -431,6 +431,12 @@ pub mod filters {
 
     /// Get the idiomatic Go rendering of a struct field name.
     pub fn field_name(nm: &str) -> Result<String, askama::Error> {
+        // Fields called 'Error' can clash with structs which implement the error
+        // interface, causing a compilation error. Suffix with _ similar to reserved
+        // keywords in var names.
+        if nm == "error" {
```

Comment:
`field_name` also applies to enum fields and record structs. I believe there shouldn't be any problems to use `Error` in an enum variant/struct. Is this intentional? You could create a separate filter `error_field_name` and use it for errors only.

Reply:
Also, maybe its beter to disallow `error` field name entirely? Having both `Error()` and `Error_` could be very confusing, especially if `Error_` is a `string`.

Reply:
Although disallowing `error` is also sub-optimal, because `error` field might make perfect sense in other languages when using multiple foreign bindings.

Reply:
I literally just put it in the easiest place I could, as I don't know this project very well :D 

I think suffixing `_` is reasonable, given we already do that for reserved Go keywords for variable names. I don't think `Error_` and `Error()` are any more confusing that `Err` and `Error()` which is what you'd typically see for nested errors natively.

I really want to apply it only in cases where the variant/struct implements the error interface, but I don't see an easy way to do that, hence the blanket rename to `Error_`.

Reply:
To separate errors and enums/structs, you could leave `field_name` function as it was before, and create a new function beside it `error_field_name` with this new `Error_` logic you implemented. Then, update [ErrorTemplate.go](https://github.com/NordSecurity/uniffi-bindgen-go/blob/main/bindgen/templates/ErrorTemplate.go) to reference `error_field_name` instead of `field_name`. This way enums/structs will continue to use the current version of `field_name`, and errors will use the new logic with `Error_` renaming.

Reply:
I have done as you asked, and this feels much better now I think!